client/question/services/EventHandlerService.js

```diff
@@ -53,20 +67,28 @@ tie.factory('EventHandlerService', [
 
     return {
       _getCurrentEventBatchLength: getCurrentEventBatchLength,
-
+      initEventHandlerServiceState: initEventHandlerServiceState,
       sendCurrentEventBatch: sendCurrentEventBatch,
 
       /**
        * Submits data to TIE's backend to create a SessionPauseEvent.
-       * @param {string} sessionId Unique ID for a user's question session.
+       * @param {string} taskId Id of the task being paused.
        *
        */
-      createSessionPauseEvent: function(sessionId) {
-        if (ServerHandlerService.doesServerExist()) {
+      createSessionPauseEvent: function(taskId) {
+        // Checking for null here in the off chance that a user pauses
+        // before the question is loaded, creating a minor race condition.
+        if (ServerHandlerService.doesServerExist() &&
+            _currentSessionId !== null &&
```

Comment:
Factor these three inequality checks into an isInitialized() method.

Also, if you want to be safe, then for other methods in EventHandlerService you can also do a "if not init-ed, throw Error" check at the outset. But that's optional.

Reply:
Okay. I'll do the isInitialized bit.